### ISSUE TITLE
Jupyterhub Scratch Write Access for Student

### ISSUE THREAD

**Issue by paselkin** [csdms/help-desk#245]
Hi Folks,
A student working with me on a Landlab project on Jupyterhub would like to share some code with me, but she doesn't seem to be able to save files to the scratch folder. Is it possible to check to make sure she has write permissions on Scratch? She does have read access: she's been able to read files I've saved on there. The student's name is Victoria Goodrich, email is tori543@uw.edu
Thanks,
Peter Selkin

**Comment by mdpiper** [comment]
@paselkin By default, all users have write access to the `scratch` directory. However, users only have read access to files they don't own; e.g., @tori543 won't be able to modify a file you own. You can [change the file permissions](https://www.redhat.com/sysadmin/linux-file-permissions-explained) on a file you own to make it writable. (Note that this makes it writable by all users on the Hub.)

**Comment by mdpiper** [comment]
@paselkin I want to be careful to emphasize that `scratch` is a really low-tech way of sharing files. It's basically a pick-up and drop-off point. Going back to an earlier conversation in #240, I think the best way for working together on a notebook is through GitHub. This is more difficult, but it may pay off in the long (even medium) run because the notebook will be under version control, so you'll have a complete history of changes to the file.

**Comment by paselkin** [comment]
Thanks. I've used GitHub before, but never with JupyterHub. Is there a way
to clone repositories directly into a JupyterHub account? Can I just open a
terminal and do a git clone https://github.com/link-to/whatever-repo, like
I do on my local machine? Or do I need to clone/checkout to my local
computer first, then upload to the JupyterHub server?

**Comment by mdpiper** [comment]
Yup! You can configure Git and GitHub for use directly on the Hub. (I actually use this often.)

It sounds like you're already comfortable doing this, but if it would help, I have a [short set of lessons](https://github.com/csdms/ivy/blob/main/lessons/git/index.md) on configuring Git/GitHub that I use when teaching classes that use the Hub.

**Comment by paselkin** [comment]
Never mind - just tried what I suggested with a test repo and it worked!
All good now: I'll just teach Tori how to use Github...
